src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,261 @@
+                new StorageBlobContainerComparer());
+            _blobsFoundFromScanOrNotification = new ConcurrentQueue<IStorageBlob>();
+        }
+
+        public void Start()
+        {
+            ThrowIfDisposed();
+            _pollLogStrategy.Start();
+        }
+
+        public void Cancel()
+        {
+            ThrowIfDisposed();
+            _pollLogStrategy.Cancel();
+            _cancellationTokenSource.Cancel();
+        }
+
+        public async Task RegisterAsync(IStorageBlobContainer container, ITriggerExecutor<IStorageBlob> triggerExecutor, CancellationToken cancellationToken)
+        {
+            // Register and Execute are not concurrency-safe.
+            // Avoiding calling Register while Execute is running is the caller's responsibility.
+            ThrowIfDisposed();
+
+            // Register all in logPolling, there is no problem if we get 2 notifications of the new blob
+            Task registerLogPolling = _pollLogStrategy.RegisterAsync(container, triggerExecutor, cancellationToken);
+
+            ICollection<ITriggerExecutor<IStorageBlob>> containerRegistrations;
+            if (!_registrations.TryGetValue(container, out containerRegistrations))
+            {
+                containerRegistrations = new List<ITriggerExecutor<IStorageBlob>>();
+                _registrations.Add(container, containerRegistrations);
+            }
+
+            containerRegistrations.Add(triggerExecutor);
+
+            if (!_lastSweepCycleStartTime.ContainsKey(container))
+            {
+                _lastSweepCycleStartTime.Add(container, DateTime.MinValue);
+            }
+
+            if (!_currentSweepCycleStartTime.ContainsKey(container))
+            {
+                _currentSweepCycleStartTime.Add(container, DateTime.MinValue);
+            }
+
+            if (!_continuationTokens.ContainsKey(container))
+            {
+                _continuationTokens.Add(container, null);
+            }
+            await registerLogPolling;
+        }
+
+        public async Task<TaskSeriesCommandResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
+
+            Task LogPollingTask = _pollLogStrategy.ExecuteAsync(cancellationToken);
+            List<IStorageBlob> failedNotifications = new List<IStorageBlob>();
+            List<Task> notifications = new List<Task>();
+
+            // Drain the background queue of blob written notifications.
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                IStorageBlob blob;
+
+                if (!_blobsFoundFromScanOrNotification.TryDequeue(out blob))
+                {
+                    break;
+                }
+
+                notifications.Add(NotifyRegistrationsAsync(blob, failedNotifications, cancellationToken));
+            }
+            Task.WhenAll(notifications).Wait();
```

Comment:
This is a blocking wait. Was that intended? if not, do:
async Task.WhenAll(notifications)